src/ua.c

```diff
@@ -1982,41 +2186,150 @@ struct sipevent_sock *uag_sipevent_sock(void)
 }
 
 
+static bool uri_match_transport(const struct uri *accu,
+		const struct uri *peeru, enum sip_transp tp)
+{
+	struct pl pl1 = PL("udp");
+	struct pl pl2 = PL_INIT;
+
+	(void)msg_param_decode(&accu->params, "transport", &pl1);
+
+	if (tp == SIP_TRANSP_NONE) {
+		/* outgoing calls */
+		pl_set_str(&pl2, sip_transp_name(uag.cfg->transp));
+		(void)msg_param_decode(&peeru->params, "transport", &pl2);
+
+		return !pl_casecmp(&pl1, &pl2);
+	}
+
+	/* incoming calls: tp of SIP message */
+	return !pl_strcasecmp(&pl1, sip_transp_name(tp));
+}
+
+
+static bool match_port(uint16_t port1, uint16_t port2, bool tls)
```

Comment:
perhaps "bool secure" is better ? what about WSS ?

Reply:
match_port() matches account port with uri port. If not set it uses the default port which is different for TLS. I will add a doxygen.

Reply:
SIP over WS/WSS also has a default port:

```
In the absence of DNS SRV resource records or an explicit port, the
   default port for a SIP URI using the "sip" scheme and the "ws"
   transport parameter is 80, and the default port for a SIP URI using
   the "sips" scheme and the "ws" transport parameter is 443.
```

Reply:
Again was thinking about the checks in uag_find_msg(). Have to discuss with my team why we added the port check. There is only (at most) one port open for each transport, anyway.